setup/bindir/cloud-set-guest-password.in

```diff
@@ -26,25 +26,58 @@
 # Modify this line to specify the user (default is root)
 user=root
 
-# Add your DHCP lease folders here
-DHCP_FOLDERS="/var/lib/dhclient/* /var/lib/dhcp3/* /var/lib/dhcp/*"
+# Add network provider variables here (default means that interface for network manager is eth0)
+NETPLAN=/etc/netplan
+IFUPDOWN=/etc/network/interfaces
+NETMAN=/etc/sysconfig/network-scripts/ifcfg-eth0
```

Comment:
@dimsijul Hi. maybe its a good idea to change NETMAN to the directory too like NETPLAN or IFUPDOWN because eth0 can change if we dont use the naming scheme.

_"If both systemd predictable interface naming (net.ifnames) and biosdevname naming schemes are disabled, network interfaces continue to use the unpredictable and potentially inconsistent ethX name originally given by the kernel."_

```shell
NETMAN=/etc/sysconfig/network-scripts
```

Reply:
Hi Sven,

I see what you mean. Yes, I'll do the change to base the detection on the folder. I think the same kind of issue would apply to the netplan command though, if the interface is not declared as eth0 in the netplan config...

Reply:
Also, I think that maybe we should do the same as the account to target  in : 

# Modify this line to specify the user (default is root)
user=root

with maybe :

# Modify this line to specify the network interface (default is eth0)
netinterface=eth0

and use that variable in netplan and nmcli commands.

What do you think ?

Reply:
@dimsijul 

> Hi Sven,
> 
> I see what you mean. Yes, I'll do the change to base the detection on the folder. I think the same kind of issue would apply to the netplan command though, if the interface is not declared as eth0 in the netplan config...

sounds good.



> Also, I think that maybe we should do the same as the account to target in :
> 
> Modify this line to specify the user (default is root)
> user=root
> 
> with maybe :
> 
> Modify this line to specify the network interface (default is eth0)
> netinterface=eth0
> 
> and use that variable in netplan and nmcli commands.
> 
> What do you think ?

maybe i thread lost... what do you mean with modify user=root and netinterface=eth0... maybe you can ... you mean to specify the first interface?

Reply:
Yes that is what I mean

Reply:
maybe grep from "ip command" or "nmcli" command the interface for your script. i think we should not change the user. so we want to change the root password.

Reply:
I did not mean to change the user, but maybe to use the same logic by specifying the interface instead of assuming it is eth0. Maybe it can be retrieved as you mention. I'll do the tests.